parts/openshift/unstable/openshiftmasterscript.sh

```diff
@@ -32,10 +32,33 @@ else
 	COCKPIT_VERSION="latest"
 fi
 
-# TODO: with WALinuxAgent>=v2.2.21 (https://github.com/Azure/WALinuxAgent/pull/1005)
-# we should be able to append context=system_u:object_r:container_var_lib_t:s0
-# to ResourceDisk.MountOptions in /etc/waagent.conf and remove this stanza.
+if grep -q ^ResourceDisk.Filesystem=xfs /etc/waagent.conf; then
+	# Bad image: docker and waagent are racing.  Try to fix up.  Leave this code
+	# until the bad images have gone away.
+	set +e
+
+	# stop docker if it hasn't failed already
+	systemctl stop docker.service
+
+	# wait until waagent has run mkfs and mounted /var/lib/docker
+	while [[ $(stat -c %D /var/lib/docker) != 811 ]]; do
+		sleep 1
+	done
+
+	# now roll us back. /var/lib/docker/* may be mounted if docker lost the
+	# race.
+	umount /var/lib/docker
+	umount /var/lib/docker/*
+
+	# disable waagent from racing again if we reboot.
+	sed -i -e '/^ResourceDisk.Format=/ s/=.*/=n/' /etc/waagent.conf
+	set -e
+fi
+
 systemctl stop docker.service
+mkfs.xfs -f /dev/sdb1
```

Comment:
Is a format required?  I know waagent is no longer going to be performing the format.

Reply:
yes - it comes in as ntfs, it needs to be xfs